File: src/main/kotlin/onl/tesseract/srp/skill/domain/model/recipe/Recipe.java
package onl.tesseract.srp.skill.domain.model.recipe;

import java.time.Duration;
import java.util.Map;

public record Recipe(
        RecipeName name,
        Slot slot,
        Map<IngredientSlot, RecipeComponent> components,
        RecipeComponent result,
        Tier tier,
        Duration duration
) {}

File: src/main/kotlin/onl/tesseract/srp/domain/skill/recipe/Recipe.kt
package onl.tesseract.srp.domain.skill.recipe

data class Recipe(
    val name: String,
    val slot: Int,
    val components : Map<Int, RecipeComponent>,
    val result: RecipeComponent,
    val tier: Int = 1
){
}

data class RecipeComponent(
    val item: ComponentWrapper,
    val quantity: Int
)